python-package/lightgbm/sklearn.py

```diff
@@ -412,6 +412,9 @@ def __init__(
         random_state: Optional[Union[int, np.random.RandomState]] = None,
         n_jobs: Optional[int] = None,
         importance_type: str = 'split',
+        use_early_stopping: bool = False,
+        validation_fraction: Optional[float] = 0.1,
+        early_stopping_round: int = 10,
```

Comment:
It shouldn't be necessary to add `n_iter_no_change` or `early_stopping_round` to the public interface of these classes.

All keyword arguments passed through `**kwargs` are appended to the `params` stored on the object and passed through to LightGBM's C++ code.

https://github.com/microsoft/LightGBM/blob/7a801f7b7b1ed8faf4cd1dbc09086c269fc51e1a/python-package/lightgbm/sklearn.py#L569

https://github.com/microsoft/LightGBM/blob/7a801f7b7b1ed8faf4cd1dbc09086c269fc51e1a/python-package/lightgbm/sklearn.py#L617-L621

`n_iter_no_change` is supported as a parameter alias for `early_stopping_rounds` in LightGBM.

https://github.com/microsoft/LightGBM/blob/7a801f7b7b1ed8faf4cd1dbc09086c269fc51e1a/src/io/config_auto.cpp#L83

And all aliases will be resolved in `lightgbm.train()`, which is called by the scikit-learn estimators' `fit()` methods.

https://github.com/microsoft/LightGBM/blob/7a801f7b7b1ed8faf4cd1dbc09086c269fc51e1a/python-package/lightgbm/sklearn.py#L842

https://github.com/microsoft/LightGBM/blob/7a801f7b7b1ed8faf4cd1dbc09086c269fc51e1a/python-package/lightgbm/engine.py#L175-L179

Can you please explain why you added this? If you just didn't realize LightGBM had this mechanism, please remove this keyword argument from all these classes (but keep passing it explicitly in the unit tests, if it's the preferred argument for `scikit-learn`).

Reply:
I did not fully understand how aliases worked in LightGBM, this comment was super helpful in unblocking me, thanks a lot for it.

I have removed `n_iter_no_change` from the public interface.

Reply:
> I have removed `n_iter_no_change` from the public interface.

I see `n_iter_no_change` in the signatures for all the estimators still. Please remove it.

This is not the first time in this PR when I've asked for something, you've said a change has been implemented, and I've spent time reviewing only to find that the code doesn't match what you've said. Please, look through the diff at https://github.com/microsoft/LightGBM/pull/5808/files carefully and compare it to the current state of the discussion threads before `@`-ing for another review.

Reply:
Ah I see now that https://github.com/microsoft/LightGBM/pull/5808#discussion_r1863324478 explains this. Let's continue there and mark this thread resolved.

For the future... you could have saved us a little time by linking these two threads together.